### ISSUE TITLE
Request: (Apple) HEIC images

### ISSUE THREAD

**Issue by iAmRenzo** [bpatrik/pigallery2#97]
Support of heic images from iPhone

**Comment by bpatrik** [comment]
Hi 

unfortunately non of the browsers are supporting that file format: 
https://caniuse.com/#feat=heif
So I cannot just add it now.

In the future I'll implement a *.* to *.jpg converter in the app, with that feature I can support heic too. (sorry deadline promises, probably will take months)

Patrik

**Comment by bpatrik** [comment]
Hi,

unfortunately its not easy to add ass HEIC not an open format. Read more here: https://github.com/lovell/sharp/issues/1105

So I cannot add this in the near future.

Best,
Patrik

**Comment by helmut72** [comment]
lovell/sharp#1105 is closed

**Comment by bpatrik** [comment]
A gave it a quick look (with sharp v0.27.0 and I still get `heif: Unsupported feature: Unsupported codec (4.3000)`
(Tested on https://github.com/tigranbs/test-heic-images/blob/master/image1.heic).

Without looking in to it in depth, this: https://github.com/lovell/sharp/issues/1105#issuecomment-631143395 suggests that works only on mac?

**Comment by presto8** [comment]
I am not that familiar with Node development, but would it be possible to use something like heif-convert (external command-line tool, does not require a Mac). If heif-convert could be used to transcode HEIF to JPG and then store the JPG files in the cache/thumbnails directory, it would be a huge benefit for those of us who have huge HEIF libraries. Thank you!

**Comment by presto8** [comment]
More info on this: libvips supports HEIF now (tested on Linux, version 8.12.1). Since sharp uses libvips, I think this feature may work if the sharp dependency is updated to the latest version?

**Comment by bpatrik** [comment]
There is an open issue about upgrading sharp to the latest version: https://github.com/bpatrik/pigallery2/issues/295

Unfortunately it is not trivial, as I would like to keep supporting arm based Docker images

**Comment by yllekz** [comment]
If it helps, "viktorstrate/PhotoView" is capable of reading the HEIC format and it runs on Pis/M1/etc. Might be able to have a look at his implementation.

**Comment by bpatrik** [comment]
There was a sharp update recently, so this request might be  unblocked. Someone need to check it.

**Comment by bpatrik** [comment]
try adding it here and see what happens: https://github.com/bpatrik/pigallery2/blob/1e8830ab09686da171b2a0a49fec5eac79943646/src/common/SupportedFormats.ts#L2

**Comment by relink2013** [comment]
Im curious about this as I would really like to give pigallery a try but nearly all of the photos I have taken in the last several years are HEIC many of them live photos with an additional .mov file.

**Comment by bpatrik** [comment]
Taking an additional look. [sharp](https://github.com/lovell/sharp) (the library used for image conversions) does not support heic out of the box: https://github.com/lovell/sharp/issues/2004#issuecomment-564104958 -> https://github.com/lovell/sharp/issues/1105#issuecomment-516010573

To my understanding Dockerfile should be updated and we need to manually install `libheif` then `libvips` then `sharp through npm`: https://github.com/lovell/sharp/issues/2004#issuecomment-565748090

Here are some Docker file samples that would be a good starting point: https://github.com/libvips/libvips/issues/1844

---------------------------------------------------

I do not own any apple product, so I'm less motivated investing in HEIC support, but happy to give guidance if anyone else would like to do change.

**Comment by martadinata666** [comment]
Looking on `https://packages.debian.org/sid/libvips-dev` seems it build with `HEIC` support, so rather than using lovell version libvips, using distro libvips seems viable 🤔

**Comment by bpatrik** [comment]
Moved the 'supportedformats' to the config, so people can easier experiment with supported extensions

**Comment by bpatrik** [comment]
Similarly to #567 and #36.

HEI* is not supported by mainstream browsers (only by safari)

![kép](https://user-images.githubusercontent.com/7512182/210665350-3fd28e01-e845-437f-b9ff-ee99cc116e6a.png)

Until that is changes, the app wont support this format either.

**Comment by yllekz** [comment]
I don't understand that chart stating Firefox doesn't support it? I can view HEIC files in Firefox just fine via other means.

**Comment by bpatrik** [comment]
I mean if you try to open a supported file (like jpg) with firefox or chrome then they will show that photo. If you try to do the same with HEIC, the browser will only offer to download that file as they cannot render them. Only safari support HEIC natively. 

So the app would need to convert HEIC (sharp support needed) to some browser supported format (jpg,webp) and only show that one. 
RAW and TIFF photos have the same issue #127. Browser cannot render them.

**Comment by elpendor** [comment]
Any news on this?

Or at least some tips on how to implement it?

**Comment by DrPeterVC** [comment]
I suspect that HEIC will gradually replace JPG as it is a superior compression format.  About a year ago my Samsung Android (A41) started to produce heic format photos (without any promoting from me - probably a system update) rather than jpg. So though Plex and FileManager + seem to be fine with displaying them it would be great it Pigallery2 could handle them too - as my gallery is now stuck in the past :-)

**Comment by bpatrik** [comment]
At the moment pigallery2 can only support those image formats that the
browsers also support natively. So you need to convince chrome and Firefox
first to implement it :)

-- Sorry for being brief, sent from my phone.

**Comment by DrPeterVC** [comment]
Many thanks for your rapid response.  I am actually mega impressed with the program.  Though I use Plex it's handling of photos is limited.  What I like about pigallery2 is it's sheer speed on pi (I have failed to get Jellyfin to work satisfactorily) and it's ability to recognise face tags (I have methodically tagged faces using various apps over the years).

I have got a very simple workaround for the heic issue - I just ran heif-convert on the directory with the photos and now all I need to do is correct the orientation with some using exiftool.

I feel sure heic will be supported by all browsers in time.

Many thanks again.

**Comment by bpatrik** [comment]
I'm happy that you like the app.

In the long run I plan to support images that browser do not support
natively (like raw photos). But I don't see that happening in the next 2-3
years.

Correcting the orientation steps seems odd to me. The app should be able to
handle it.

-- Sorry for being brief, sent from my phone.

**Comment by v-marinkov** [comment]
I love the ability of Pigallery to read index.md files per folder. I'd been looking for a piece of software that replicates Google Photo's blog-type paragraphs. Unfortunately, the lack of HEIC support is the only dealbreaker for me.

**Comment by DrPeterVC** [comment]
I think browsers will eventually have to include HEIC and this will happen automatically in time.  But in the meantime I get round it by adding heif-convert to run on newly downloaded files (watch for the orientation flag though).  My intention is that all the generated JPGs can be removed once native support appears!  This is still my favourite photo gallery tool, by far, for showing photos to friends and family.  I am in the process of making it available through SWAG and possibly changing the config to User by default and having authelia as a more robust login process.  The inability to change passwords through the GUI is a minor irritation.

**Comment by moneytoo** [comment]
>I think browsers will eventually have to include HEIC and this will happen automatically in time.

I wouldn't be so optimistic: https://bugs.chromium.org/p/chromium/issues/detail?id=843310

>Support .HEIC: WontFix (Closed)

>No plans. Safari doesn't even support HEIC AFAIK: https://caniuse.com/?search=heic - We'd need a software decoder to add support in Chrome, so it's pretty unlikely that we'd add support for HEIC.

**Comment by v-marinkov** [comment]
> I have got a very simple workaround for the heic issue - I just ran heif-convert on the directory with the photos and now all I need to do is correct the orientation with some using exiftool.

Are there any quality sacrifices doing this? Do you still keep the original .heic files along the .jpg?

**Comment by ReessKennedy** [comment]
PROPOSAL: Since Pigallery2 creates thumbs for everything you see in a web browser (in Webm format) then why cant it just generate the Webm files from heic ... Seems there are many libraries that do this: https://github.com/alexcorvi/heic2any

Then if people want to download originals they can download the HEIC files othrwise they are just looking at the web-compatible thumbnails!

Pros: 
- Your actual library can stay small and use the latest tech
- You needn't run through hoops to pre-convert to JPEG just to make it work for PiGallery

Amazing work on this @bpatrik! Thanks for all you do. I get perhaps you use Android so maybe this isn't a personal use case improvement but would be a major win for iPhone users. At first I was annoyed with Apple for creating this new file format but when I realized it's 60% smaller file size and superior in quality I was sold! ... Realizing that almost every photo gallery app generates thumbnails I decided that flexible thumbnail generation seems like the best way to get around this issue.

**Comment by ReessKennedy** [comment]
As additional FYI, PhotoView and PhotoPrism do something like this but they both create larg-ish Jpegs thumbs and I am thinking Pigallery is fine with just create very small Webp thumbs to keep things minimal! ... As an example, this is the Photoview sidebar showing that it basically create a similarly-sized Jpeg of the HEIC file I uploaded. 

<img width="596" alt="Photo view sizing   " src="https://github.com/bpatrik/pigallery2/assets/3970267/6cc92646-4803-4eb4-9c87-ab6c9dfe60b7">

**Comment by bpatrik** [comment]
Hi

There is already a thread around this question somewhere. Basically the
conclusion here is that lib(sharp) that app uses to generate thumbnails
support heic, but libvips(the underlying lib) should be manually built in
the docker file with the support of heic.

Same for DNG raw photos (an other separate thread for that)



-- Sorry for being brief, sent from my phone.

**Comment by ReessKennedy** [comment]
AMAZING! ... Sorry I didn't catch this. I will try it. Amazing work, again.

**Comment by martadinata666** [comment]
Long story short about the "alpine" image build that still does not correctly read HEIF/HEIC. The alpine "vips-dev" package doesn't bother to install "vips-heif" so the sharp module still can't read heif/heic correctly when on Debian based "libvips-dev" that I experienced, will also pull "libheif". So yeah, we just need add "vips-heif" to "Dockerfile.alpine".

```
#-----------------BUILDER-----------------
#-----------------------------------------
FROM node:18-alpine3.17 AS builder
RUN apk add --update-cache --repository https://alpine.global.ssl.fastly.net/alpine/v3.17/community/ \
  python3 build-base sqlite-dev sqlite-libs vips-dev vips-heif fftw-dev gcc g++ make libc6-compat && ln -snf /usr/bin/python3 /usr/bin/python
COPY pigallery2-release /app
WORKDIR /app
RUN npm install --unsafe-perm
RUN mkdir -p /app/data/config && \
    mkdir -p /app/data/db && \
    mkdir -p /app/data/images && \
    mkdir -p /app/data/tmp


#-----------------MAIN--------------------
#-----------------------------------------
FROM node:18-alpine3.17 AS main
WORKDIR /app
ENV NODE_ENV=production \
    # overrides only the default value of the settings (the actualy value can be overwritten through config.json)
    default-Database-dbFolder=/app/data/db \
    default-Media-folder=/app/data/images \
    default-Media-tempFolder=/app/data/tmp \
    default-Extensions-folder=/app/data/config/extensions \
    # flagging dockerized environemnt
    PI_DOCKER=true

EXPOSE 80
RUN apk add --update-cache --repository https://alpine.global.ssl.fastly.net/alpine/v3.17/community/ \
    vips vips-cpp ffmpeg vips-heif
COPY --from=builder /app /app
VOLUME ["/app/data/config", "/app/data/db", "/app/data/images", "/app/data/tmp"]

# Run build time diagnostics to make sure the app would work after build is finished
RUN ["node", "./src/backend/index", "--expose-gc",  "--run-diagnostics", "--config-path=/app/diagnostics-config.json"]
HEALTHCHECK --interval=40s --timeout=30s --retries=3 --start-period=60s \
  CMD wget --quiet --tries=1 --no-check-certificate --spider \
  http://localhost:80/heartbeat || exit 1

# after a extensive job (like video converting), pigallery calls gc, to clean up everthing as fast as possible
# Exec form entrypoint is need otherwise (using shell form) ENV variables are not properly passed down to the app
ENTRYPOINT ["node", "./src/backend/index", "--expose-gc",  "--config-path=/app/data/config/config.json"]

```

**Comment by ReessKennedy** [comment]
I got it to work following the instructions from @martadinata666 and rebuilding Pigallery2 using the edge-alpine image ... https://hub.docker.com/layers/bpatrik/pigallery2/edge-alpine/images/sha256-e36eee2f3d9ee026ff6394dfc4d69989bebc42d6ecc4c7062407c9a722a3649e?context=explore

So using `image: bpatrik/pigallery2:edge-alpine`

I had initially just been using `image: bpatrik/pigallery2:latest` in my docker compose. 

Once I rebuilt using this I just went into config and added heic and heif to the list of supported files and then reran a bunch of the jobs and refreshed and BOOM! Success. 

Hopefully this helps others who wish to not have to convert all their iphone or heic photos in advance.

**Comment by bpatrik** [comment]
@martadinata666 Do you know if this works for all 3 platforms now? (lost track with the PRs)

If yes, we should add `heic` support to the code itself: https://github.com/bpatrik/pigallery2/blob/4352ff89b76bda8c803c670c158fd51749f568dc/src/common/config/public/ClientConfig.ts#L1257

**Comment by martadinata666** [comment]
Current state, in simple terms and in term of thumbnail generation
:edge-alpine 👍🏼 
:edge-buster 👎🏼 
:edge-bullseye 👎🏼 
:bookworm 👍🏼 (not exist yet, but tested locally)

**Comment by PaddyPat** [comment]
The latest update 7hrs ago for bpatrik/pigallery2:edge-alpine doesn’t work. How could we use the previous version of bpatrik/pigallery2:edge-alpine

**Comment by martadinata666** [comment]
If the square moving/animating it should be generating thumbnails. I had some personal HEIF images, it works. On heic taken from https://github.com/tigranbs/test-heic-images it also works.

Any log given? My logs sample as pigallery rendering
```
pigallery2-test-pigallery2-1  | 3/8/2024, 7:20:15 AM[VERBS] GET /pgapi/gallery/content/HEIC 200 180ms
pigallery2-test-pigallery2-1  | 3/8/2024, 7:20:15 AM[VERBS] GET /pgapi/share/list/HEIC 304 5ms
pigallery2-test-pigallery2-1  | 3/8/2024, 7:20:15 AM[VERBS] GET /pgapi/gallery/content/HEIC/README.md 304 6ms
pigallery2-test-pigallery2-1  | 3/8/2024, 7:20:15 AM[SILLY] [SharpRenderer] rendering photo:/app/data/images/HEIC/image2.heic, size:320
pigallery2-test-pigallery2-1  | 3/8/2024, 7:20:15 AM[SILLY] [SharpRenderer] rendering photo:/app/data/images/HEIC/image3.heic, size:320
pigallery2-test-pigallery2-1  | 3/8/2024, 7:20:15 AM[SILLY] [SharpRenderer] rendering photo:/app/data/images/HEIC/image4.heic, size:320
pigallery2-test-pigallery2-1  | 3/8/2024, 7:20:15 AM[SILLY] [SharpRenderer] rendering photo:/app/data/images/HEIC/image5.png, size:540
pigallery2-test-pigallery2-1  | 3/8/2024, 7:20:18 AM[VERBS] GET /pgapi/gallery/content/HEIC/image2.heic/320 200 3040ms
pigallery2-test-pigallery2-1  | 3/8/2024, 7:20:18 AM[VERBS] GET /pgapi/gallery/content/HEIC/image3.heic/320 200 3170ms
pigallery2-test-pigallery2-1  | 3/8/2024, 7:20:19 AM[VERBS] GET /pgapi/gallery/content/HEIC/image4.heic/320 200 4174ms
pigallery2-test-pigallery2-1  | 3/8/2024, 7:20:19 AM[VERBS] GET /pgapi/gallery/content/HEIC/image5.png/540 200 4328ms
pigallery2-test-pigallery2-1  | 3/8/2024, 7:20:22 AM[VERBS] GET /pgapi/gallery/content/ 200 35ms
pigallery2-test-pigallery2-1  | 3/8/2024, 7:20:22 AM[VERBS] GET /pgapi/share/list/ 304 6ms
pigallery2-test-pigallery2-1  | 3/8/2024, 7:20:22 AM[SILLY] [SharpRenderer] rendering photo:/ap
```
```
pigallery2-test-pigallery2-1  | 3/8/2024, 7:20:26 AM[VERBS] GET /pgapi/gallery/content/HEIF 200 244ms
pigallery2-test-pigallery2-1  | 3/8/2024, 7:20:26 AM[VERBS] GET /pgapi/share/list/HEIF 304 5ms
pigallery2-test-pigallery2-1  | 3/8/2024, 7:20:26 AM[SILLY] [SharpRenderer] rendering photo:/app/data/images/HEIF/IMG_5438-1.heif, size:320
pigallery2-test-pigallery2-1  | 3/8/2024, 7:20:26 AM[SILLY] [SharpRenderer] rendering photo:/app/data/images/HEIF/IMG_5436-1.heif, size:540
pigallery2-test-pigallery2-1  | 3/8/2024, 7:20:26 AM[SILLY] [SharpRenderer] rendering photo:/app/data/images/HEIF/IMG_5439-1.heif, size:320
pigallery2-test-pigallery2-1  | 3/8/2024, 7:20:30 AM[VERBS] GET /pgapi/gallery/content/HEIF/IMG_5438-1.heif/320 200 4416ms
pigallery2-test-pigallery2-1  | 3/8/2024, 7:20:30 AM[VERBS] GET /pgapi/gallery/content/HEIF/IMG_5439-1.heif/320 200 4602ms
pigallery2-test-pigallery2-1  | 3/8/2024, 7:20:30 AM[VERBS] GET /pgapi/gallery/content/HEIF/IMG_5436-1.heif/540 200 4680ms
pigallery2-test-pigallery2-1  | 3/8/2024, 7:20:36 AM[DEBUG] GET /heartbeat 200 15ms
pigallery2-test-pigallery2-1  | 3/8/2024, 7:20:39 AM[SILLY] [SharpRenderer] rendering photo:/app/data/images/HEIF/IMG_5439-1.heif, size:2160
pigallery2-test-pigallery2-1  | 3/8/2024, 7:20:43 AM[VERBS] GET /pgapi/gallery/content/HEIF/IMG_5439-1.heif/2160 200 4619ms
pigallery2-test-pigallery2-1  | 3/8/2024, 7:21:16 AM[DEBUG] GET /heartbeat 200 4ms
pigallery2-test-pigallery2-1  | 3/8/2024, 7:21:30 AM[SILLY] [SharpRenderer] rendering photo:/app/data/images/RAW/field.ARW, size:320
pigallery2-test-pigallery2-1  | 3/8/2024, 7:21:30 AM[WARN_] Handled error:
pigallery2-test-pigallery2-1  | ErrorDTO {
pigallery2-test-pigallery2-1  |   code: 32,
pigallery2-test-pigallery2-1  |   message: 'Error during generating thumbnail: /app/data/images/RAW/field.ARW',
pigallery2-test-pigallery2-1  |   details: 'Error: Input file contains unsupported image format',
pigallery2-test-pigallery2-1  |   request: { method: '', url: '' }
pigallery2-test-pigallery2-1  | }
```

**Comment by bpatrik** [comment]
@PaddyPat can you try a DB reset?

If it doesn't solve it, can provide debug logs from the server, and from the browser maybe also some screenshots to see where it stops.

**Comment by PaddyPat** [comment]
running pigallery2 with mariadb.
when I add new heic images, I see the black quare and in logfiles:

pigallery2-db  | 2024-03-08 12:50:14 0 [Note] /usr/bin/mysqld: ready for connections.
pigallery2-db  | Version: '10.5.11-MariaDB-log'  socket: '/run/mysqld/mysqld.sock'  port: 3306  MariaDB Server
pigallery2     | 3/8/2024, 11:50:15 AM[INFO_][server] running diagnostics...
pigallery2     | 3/8/2024, 11:50:16 AM[INFO_][JobManager] Running job schedules
pigallery2     | 3/8/2024, 11:50:16 AM[INFO_][server] Listening on port 80
pigallery2     | 3/8/2024, 11:50:25 AM[WARN_] Handled error:
pigallery2     | ErrorDTO {
pigallery2     |   code: 31,
pigallery2     |   message: 'Error during listing the directory',
pigallery2     |   details: QueryFailedError: ER_BAD_FIELD_ERROR: Unknown column 'media.metadataCreationdateoffset' in 'field list'
pigallery2     |       at Query.<anonymous> (/app/node_modules/typeorm/driver/mysql/MysqlQueryRunner.js:158:37)
pigallery2     |       at Query.<anonymous> (/app/node_modules/mysql/lib/Connection.js:526:10)
pigallery2     |       at Query._callback (/app/node_modules/mysql/lib/Connection.js:488:16)
pigallery2     |       at Sequence.end (/app/node_modules/mysql/lib/protocol/sequences/Sequence.js:83:24)
pigallery2     |       at Query.ErrorPacket (/app/node_modules/mysql/lib/protocol/sequences/Query.js:92:8)
pigallery2     |       at Protocol._parsePacket (/app/node_modules/mysql/lib/protocol/Protocol.js:291:23)
pigallery2     |       at Parser._parsePacket (/app/node_modules/mysql/lib/protocol/Parser.js:433:10)
pigallery2     |       at Parser.write (/app/node_modules/mysql/lib/protocol/Parser.js:43:10)
pigallery2     |       at Protocol.write (/app/node_modules/mysql/lib/protocol/Protocol.js:38:16)
pigallery2     |       at Socket.<anonymous> (/app/node_modules/mysql/lib/Connection.js:88:28)
pigallery2     |       at Socket.<anonymous> (/app/node_modules/mysql/lib/Connection.js:526:10)
pigallery2     |       at Socket.emit (node:events:517:28)
pigallery2     |       at addChunk (node:internal/streams/readable:368:12)
pigallery2     |       at readableAddChunk (node:internal/streams/readable:341:9)
pigallery2     |       at Readable.push (node:internal/streams/readable:278:10)
pigallery2     |       at TCP.onStreamRead (node:internal/stream_base_commons:190:23) {
pigallery2     |     query: 'SELECT `directory`.`id` AS `directory_id`, `directory`.`name` AS `directory_name`, `directory`.`path` AS `directory_path`, `directory`.`lastModified` AS `directory_lastModified`, `directory`.`lastScanned` AS `directory_lastScanned`, `directory`.`mediaCount` AS `directory_mediaCount`, `directory`.`oldestMedia` AS `directory_oldestMedia`, `directory`.`youngestMedia` AS `directory_youngestMedia`, `directory`.`validCover`

**Comment by bpatrik** [comment]
Please reset your DB that will fix it.

-- Sorry for being brief, sent from my phone.

**Comment by benhedrington** [comment]
@bpatrik I am using edge-alpine and working well for HEIC. What will be the event or timing that transitions this functionality to latest-alpine or other such more standard image?

**Comment by bpatrik** [comment]
I have no timeline yet. I would like to get the extensions to a relatively stable state for the next release,but I got quite busy with life recently.

To give some ETA: if it won't happen in May then maybe only in September

**Comment by benhedrington** [comment]
> I have no timeline yet. I would like to get the extensions to a relatively stable state for the next release,but I got quite busy with life recently.
> 
> To give some ETA: if it won't happen in May then maybe only in September

I can appreciate that and have lived through those times. Appreciate the rough ETA.

**Comment by rhatguy** [comment]
I switched over to edge alpine last night, added heic to the supported types, and its working with images shot on my Oneplus 12.  The thumbnails appear to be square, so they look stretched.  Info on the picture is reporting that its a 512x512 pixel picture, which I assume has something to do with the stretched thumbnails.  Is that expected at this point?

**Comment by bpatrik** [comment]
>  Info on the picture is reporting that its a 512x512 pixel picture

The issues is with the Metadataloader reading the wrong dimension. There was a significant refactor there recently. I guess heic was not properly tested.

**Comment by rubpa** [comment]
> Please reset your DB that will fix it.

Hi! Trying out pigallery2 for the first time and I've HEIC images. I already ran once with `bpatrik/pigallery2:latest`. Now I modified `docker-compose.yml` and replaced the image with `bpatrik/pigallery2:edge-alpine`. However, the heic files don't show up. I guess I need to reset the db or do more changes. Kindly advise.

**Comment by rhatguy** [comment]
There is a button in the settings for PiGallery2 called "Run now: Gallery reset".  This will effectively cause PiGallery2 to reindex everything on the next index run.  Separately, you could also just delete the db itself from wherever you have docker storing it.  There should be no data in the DB that can't be re-created by re-indexing.

**Comment by rubpa** [comment]
Thank rhatguy! Here's what worked. I could not find supported formats in the Settings UI. Hence modified `config.json`, `Photo` -> `supportedFormats` and added heic there. Then in Settings -> Indexing -> Run now: Gallery reset. The heic photos started showing up. As you have pointed out earlier, the thumbnails are square but the photos load correctly.

Using Firefox on Windows, the heic photos do show up even when clicked. I understand that the webp format is loaded in this view. Only when I zoom in (double click), the heic seems to be transferred and it doesn't show the photo. However, pinch zoom photo does show on Safari on the iPhone I have.

**Comment by bpatrik** [comment]
I guess FF doe snot support heic. When you zoom the original is loaded, otherwise webp. 
Photos are square because there is an issue in the metadaloader to determine its exact dimension, so a default 1x1 is used (you can double check that in the info sidebar when you preview the photo.

**Comment by bpatrik** [comment]
You can reset by hitting this button: 
![kép](https://github.com/user-attachments/assets/e442144c-70f7-4ada-b0ad-2c256e44e396)

Probably need to change the settings to be advanced or similar on the top.

**Comment by cadeon** [comment]
Not working for me unless I zoom all the way into the image, docker, edge-alpine and latest. 

    'heif: Error while loading plugin: Support for this compression format has not been built in (11.6003)',

Weirdly the broken thumbnails seem to be the right ratios, not just squares.

**Comment by rhatguy** [comment]
Please see issue #965.  There is already some work going on for this.  At the moment PiGallery2 doesn’t support HEIC at all due to upstream packages dropping support for the format.

**Comment by bpatrik** [comment]
I'm closing this as once it worked on alpine and the current workstream is on #965

**Comment by bpatrik** [comment]
this should be fixed now
